model/host/db.go

```diff
@@ -457,3 +457,35 @@ func QueryWithFullTaskPipeline(match bson.M) []bson.M {
 		},
 	}
 }
+
+type InactiveHostCounts struct {
+	HostType string `bson:"_id"`
+	Count    int    `bson:"count"`
+}
```

Comment:
the use of this in the job is (essentially) as a map of provider strings to counts. seems like we could flatten it in the db rather than iterating over the results at the end.

Reply:
The iterating was to merge anything that wasn't static. However I think that separating those out is actually useful, so I took out the iterating